packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1053,6 +1069,69 @@ export class TransactionController extends BaseControllerV1<
     this.hub.emit(`${transactionMeta.id}:speedup`, newTransactionMeta);
   }
 
+  startTrackingByNetworkClientId(networkClientId: NetworkClientId) {
+    const networkClient = this.getNetworkClientById(networkClientId);
+    // track using tracking map
+    this.trackingMap.set(networkClientId, new Set());
+    const nonceTracker = new NonceTracker({
+      provider: networkClient.provider as any,
+      blockTracker: networkClient.blockTracker,
+      getPendingTransactions:
+        this.getNonceTrackerPendingTransactions.bind(this),
+      getConfirmedTransactions: this.getNonceTrackerTransactions.bind(
+        this,
+        TransactionStatus.confirmed,
+      ),
+    });
+    const incomingTransactionHelper = new IncomingTransactionHelper({
+      blockTracker: networkClient.blockTracker,
+      getCurrentAccount: this.getSelectedAddress,
+      getLastFetchedBlockNumbers: () => this.state.lastFetchedBlockNumbers,
+      getNetworkState: this.getNetworkState, // TODO: fake this via networkClient
+      isEnabled: () => true,
+      queryEntireHistory: true,
+      remoteTransactionSource: new EtherscanRemoteTransactionSource({
```

Comment:
we should be able to only make 1 `EtherscanRemoteTransactionSource` since it deals with chainIds itself